Bio/bgzf.py

```diff
@@ -593,30 +675,37 @@ def __init__(self, filename=None, mode="r", fileobj=None, max_cache=100):
         if filename and fileobj:
             raise ValueError("Supply either filename or fileobj, not both")
         # Want to reject output modes like w, a, x, +
-        if mode.lower() not in ("r", "tr", "rt", "rb", "br"):
+        if not _is_read_mode(mode.lower()):
             raise ValueError(
                 "Must use a read mode like 'r' (default), 'rt', or 'rb' for binary"
             )
         # If an open file was passed, make sure it was opened in binary mode.
-        if fileobj:
+        if fileobj is not None:
             if fileobj.read(0) != b"":
                 raise ValueError("fileobj not opened in binary mode")
             handle = fileobj
         else:
+            assert filename is not None
             handle = _open(filename, "rb")
         self._text = "b" not in mode.lower()
+        self._empty_buffer: _BufferT
+        self._newline: _BufferT
         if self._text:
-            self._newline = "\n"
+            self._empty_buffer = ""  # type: ignore[assignment]
+            self._newline = "\n"  # type: ignore[assignment]
         else:
-            self._newline = b"\n"
+            self._empty_buffer = b""  # type: ignore[assignment]
+            self._newline = b"\n"  # type: ignore[assignment]
         self._handle = handle
         self.max_cache = max_cache
-        self._buffers = {}
-        self._block_start_offset = None
-        self._block_raw_length = None
+        self._buffers: dict[int, tuple[_BufferT, int]] = {}
+        # Is there a safe default for a non-`None` integer?
+        self._block_start_offset: int = None  # type: ignore[assignment]
```

Comment:
Optional[int] perhaps?

We can't used zero here.

Reply:
Yes, I have had to introduce occasional ``assert isintance(...)`` lines in my own project. As you say, largely visual noise :(